src/ast-parser.y

```diff
@@ -28,16 +28,26 @@
 
 #define INVALID_VAR_INDEX (-1)
 
-/* the default value for YYMAXDEPTH is 10000, which can be easily hit since our
-   grammar is right-recursive.
+#define RELOCATE_STACK(type, array, stack_base, old_byte_size, new_size)      \
+  do {                                                                        \
+    if ((stack_base) == (array)) {                                            \
+      (stack_base) = (type*)wabt_alloc((new_size) * sizeof(*(stack_base)));   \
+      memcpy((stack_base), (array), old_byte_size);                           \
```

Comment:
Ensure that memcpy is legal, i.e. assert that `std::is_trivially_copyable<Foo>` or however that bit is spelled.